property_tests/tests/record/test/test.ts

```diff
@@ -0,0 +1,174 @@
+
+        const body = generateBody(paramRecords, returnRecord);
+
+        const test = generateTest(functionName, paramRecords, returnRecord);
+
+        return {
+            imports,
+            candidTypeDeclarations,
+            functionName,
+            paramNames,
+            paramCandidTypes,
+            returnCandidType,
+            body,
+            test
+        };
+    });
+
+runPropTests(RecordTestArb);
+
+function generateBody(
+    paramRecords: Candid<Record>[],
+    returnRecord: Candid<Record>
+): string {
+    const paramsAreRecords = paramRecords
+        .map((record, index) => {
+            const paramName = `param${index}`;
+            const fieldsCount = Object.keys(record.value).length;
+
+            const paramIsObject = `typeof ${paramName} === 'object'`;
+            const paramHasCorrectNumberOfFields = `Object.keys(${paramName}).length === ${fieldsCount}`;
+            const throwError = `throw new Error('${paramName} must be a Record');`;
+
+            return `if (!(${paramIsObject} && ${paramHasCorrectNumberOfFields})) ${throwError}`;
+        })
+        .join('\n');
+
+    const paramsCorrectlyOrdered = paramRecords
+        .map((record, index) => {
+            const paramName = `param${index}`;
+
+            const fieldNamesMatch = Object.entries(record.value)
+                .map(([fieldName, _]) => {
+                    return `Object.keys(${paramName}).includes('${fieldName}')`;
+                })
+                .join(' && ');
+
+            const throwError = `throw new Error('${paramName} is incorrectly ordered')`;
+
+            if (Object.entries(record.value).length === 0) {
+                return `if (Object.keys(${paramName}).length !== 0) ${throwError}`;
+            }
+
+            return `if (!(${fieldNamesMatch})) ${throwError}`;
+        })
+        .join('\n');
+
+    const returnStatement =
+        paramRecords.length === 0 ? returnRecord.src.valueLiteral : `param0`;
+
+    return `
+        ${paramsAreRecords}
+
+        ${paramsCorrectlyOrdered}
+
+        return ${returnStatement};
+    `;
+}
+
+function generateTest(
+    functionName: string,
+    paramRecords: Candid<Record>[],
+    returnRecord: Candid<Record>
+): Test {
+    return {
+        name: `record ${functionName}`,
+        test: async () => {
+            const actor = getActor('./tests/record/test');
+
+            const result = await actor[functionName](
+                ...paramRecords.map((record) => record.value)
+            );
+
+            // This built in equals will handle types like principal without
+            // any additional work. Do this first. If it fails, move on to the
+            // more robust check that will give us clues as to why it failed
+            if (returnRecord.equals(result, returnRecord.value)) {
+                return { Ok: true };
+            }
+            return recordsAreEqual(result, returnRecord.value);
+        }
+    };
+}
+
+function recordsAreEqual(
```

Comment:
Let's not create custom equals things like this, instead let's see if we can print out the two values that aren't equal

Reply:
Documented on #1390. Will tackle later.